screens/Menu/CourseSchedule.js

```diff
@@ -23,53 +24,143 @@ import PropTypes from "prop-types";
+        });
+        let resp = await calendarEvents.json();
+        let filteredList = getFilteredGoogleCalendarEvents(resp);
+        let result = filteredList.filter(element => element.summary)
+        setCalendarEventsList(result);
+        setRefresh(false);
+    }
+
+    const getFilteredGoogleCalendarEvents = (resp) => {
+        var filteredList = resp.items.map(element => {
+            let elementDate = new Date (element.start.date);
+            if(elementDate > currentDate){
+                return { id: element.id, summary: element.summary, description: element.description, location: element.location };
+            }
+            else{
+                return {};
+            }
+        });
+        return filteredList;
+    };
+
+    const handleRefresh = () => {
+        setRefresh(true);
+        getCalendarEvents();
+    }
+
+    const CourseScheduleScreen = true;
+
+    /**
      * The method will slide the side menu from the right side of the screen
      * @param  {} =>{props.navigation.openDrawer(
-     */
+    */
     const goToMenu = () => {
         props.navigation.openDrawer();
     };
 
-    
-    /**
-     * The method will let us navigate to the CourseSchedule screen
-     * @param  {} =>{props.navigation.navigate("CourseScheduleDetails"
-     */
-    const goToCourseSchedule = () => {
-        props.navigation.navigate("CourseScheduleDetails");
-    };
+    const goToDoubleSearch = (item) => {
+        props.navigation.navigate("DoubleSearch", { CourseScheduleScreen: CourseScheduleScreen, CourseScheduleLocation: item.location });
+    }
 
-    // TODO: Static data for now just to layout the SectionList 
-    const data = [{
-        title: "Wed, Feb 5",
-        data: ["SOEN 357 Lecture", "SOEN 357 Tutorial", "SOEN 345 Lecture"
-        ]
-    }];
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            getCalendarId();
+            getAccessToken();
+            getSwitchValue();
+        }, 1);
+        getCalendarEvents();
+        return () => clearInterval(intervalId);
+    }, [selectedCalendarId, accessToken]);
 
+    if (switchVal == "true" && calendarEventsList) {
     return (
-        <View style={styles.container}>
-            <View style={styles.menuButtonContainer}>
-                <TouchableOpacity style={styles.menuButton} onPress={goToMenu}>
-                    <Feather name="menu" style={styles.icon} />
-                </TouchableOpacity>
+            <View style={styles.container}>
+                <View style={styles.menuButtonContainer}>
+                    <TouchableOpacity style={styles.menuButton} onPress={goToMenu}>
+                        <Feather name="menu" style={styles.icon} />
+                    </TouchableOpacity>
+                </View>
+                <Text style={styles.mainLabel}>My Course Schedule</Text>
+                <View style={styles.scrollTextContainer}>
+                    {(switchVal == "true" && calendarEventsList) &&
```

Comment:
extend this conditional rendering with the following: 
`{(switchVal == "true" && calendarEventsList.length > 0) &&`

Reply:
Its the same thing no ?

Reply:
Nop it's not the same thing!